src/app/calendar/parcelCalendarFunctions.ts

```diff
@@ -1,44 +1,33 @@
 import supabase, { Schema } from "@/supabase";
 import { CalendarEvent } from "@/components/Calendar/Calendar";
 
-const COLLECTION_DURATION_MS = 30 * 60 * 1000;
+export type ParcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
 
 export interface LocationColorMap {
     [location: string]: { color: string; text: string };
 }
 
-export interface ClientMap {
-    [primary_key: string]: Schema["clients"];
-}
-
-export const getClientMap = async (): Promise<ClientMap> => {
-    const response = await supabase.from("clients").select();
-    const clients = response.data ?? [];
-    const clientMap: ClientMap = {};
+const COLLECTION_DURATION_MS = 30 * 60 * 1000;
 
-    clients.forEach((client) => {
-        clientMap[client.primary_key] = client;
-    });
+export const getParcelsWithCollectionDate = async (): Promise<ParcelWithClientName[]> => {
+    const { data } = await supabase
+        .from("parcels")
+        .select("*, clients ( full_name )")
+        .not("collection_datetime", "is", null);
 
-    return clientMap;
-};
+    if (data == null) {
```

Comment:
potentially also `const {data, error} = await ...` and test if `error` is not null - check the repo for examples of this elsewhere!

Reply:
done